### PR TITLE
Move shop to a modal

### LINKED ISSUE #292: In a shop, the 'Return to town' is more prominent than the buy button.
I am an idiot and accidentally pressed it.
Maybe hide it a bit more, please excuse my photoshop skills
image

### PR DESCRIPTION
**Pull request by RedSparr0w**
closes #292
image

### CODE REVIEW

**Comment by Ishadijcks** [comment]
Holding off on this pr until #307 is merged.

**Comment by Aegyo** [comment]
@RedSparr0w Could you fix the merge conflict on this please?

**Comment by RedSparr0w** [comment]
@Aegyo Fixed the merge conflicts,
Also updated so that the shop modal closes when a key item is purchased.
And renamed the `shopView` to `shopModal` and moved the import position in the `index`

File: src/scripts/shop/ShopHandler.ts [hunk 1]
```diff
@@ -6,15 +6,12 @@ class ShopHandler {
...
         ShopHandler.amount(1);
-        Game.gameState(GameConstants.GameState.idle);
         this.shopObservable(shop);
 
         for (let i = 0; i < shop.items().length; i++) {
             let item: Item = shop.items()[i];
             item.price(Math.round(item.basePrice * player.itemMultipliers[item.name()]));
         }
-
-        Game.gameState(GameConstants.GameState.shop);
```
Comment (Ishadijcks) [inline src/scripts/shop/ShopHandler.ts]: The whole shop gamestate can be removed now right?

**Pull request closed without merging**